cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/validator/privatedns/AzureExistingPrivateDnsZoneValidatorService.java

```diff
@@ -0,0 +1,42 @@
+package com.sequenceiq.cloudbreak.cloud.azure.validator.privatedns;
+
+import java.security.InvalidParameterException;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.cloud.azure.AzurePrivateDnsZoneServiceEnum;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClient;
+import com.sequenceiq.cloudbreak.validation.ValidationResult;
+
+@Service
+public class AzureExistingPrivateDnsZoneValidatorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureExistingPrivateDnsZoneValidatorService.class);
+
+    @Inject
+    private AzurePrivateDnsZoneValidatorService azurePrivateDnsZoneValidatorService;
+
+    public ValidationResult.ValidationResultBuilder validate(AzureClient azureClient, String networkResourceGroupName,
+            String networkName, Map<AzurePrivateDnsZoneServiceEnum, String> serviceToPrivateDnsZoneId, ValidationResult.ValidationResultBuilder resultBuilder) {
+        serviceToPrivateDnsZoneId.forEach((service, privateDnsZoneId) -> {
+            try {
+                azurePrivateDnsZoneValidatorService.existingPrivateDnsZoneNameIsSupported(service, privateDnsZoneId, resultBuilder);
+                azurePrivateDnsZoneValidatorService.privateDnsZoneExists(azureClient, privateDnsZoneId, resultBuilder);
+                azurePrivateDnsZoneValidatorService.privateDnsZoneConnectedToNetwork(azureClient, networkResourceGroupName, networkName, privateDnsZoneId,
+                        resultBuilder);
+            } catch (InvalidParameterException e) {
```

Comment:
Yes, I was thinking about it also myself. It then would need to give a return value if the azureResourceId is valid, so that the other methods are not executed. The method's return value is the validationResoultBuilder. I then would need to change the return value to a boolean to express that the resourceId is not valid. Would that be ok for you?

Reply:
If we go this way, then this is the case for all the subsequent calls, each needs to be called only if the previous one did not have
`resultBuilder.build().hasError() == true`

I am not against running all the validations at once and have multiple errors combined in the end instead.

Reply:
Here I wanted to avoid parsing an invalid azure resource id, because in that case there are no more validations needed for that private DNS zone, and validation can proceed with the next private DNS zone. I am also fond of doing all the validation that make sense and return a list of problems instead of returning errors one by one.

So, throwing an exception does exactly this, not entering validations that are not needed.

Reply:
I get your point, this exception is thrown in each of these, so consider this comment invalid.